BridgeSDK/BridgeAPI/SBBUploadManager.m

```diff
@@ -544,42 +547,109 @@ - (void)uploadFileToBridge:(NSURL *)fileUrl contentType:(NSString *)contentType
 - (void)kickOffUploadForFile:(NSString *)filePath uploadRequestJSON:(NSDictionary *)uploadRequestJSON
 {
     // this starts the process by downloading the Bridge UploadSession.
+    // first, make sure we get rid of any existing UploadSession for this file, in case this is a retry.
+    [self setUploadSessionJSON:nil forFile:filePath];
```

Comment:
Not sure I understand the purpose of this. why remove the object from the dictionary rather than just replace with the new key? Wouldn't it be simpler and safer to just `@synchronize` the method `setUploadRequestJSON:forFile:`?